src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,215 @@
+            {
+                string response = await m.Content.ReadAsStringAsync();
+
+                var assets = JsonConvert.DeserializeObject<GHUpdate>(JsonObject.Parse(response)[0].ToString());
+                return assets;
+            }
+            else
+                throw new Exception("Could not connect - error " + m.StatusCode);
+        }
+        
+        public static async Task<bool> CheckForUpdate()
+        {
+            GHUpdate assets = await GetGithubUpdates();
+
+            //parse tag name to find the current and latest version
+            //finding the format of xx.yy.zz
+            string newestVersionSplit = Regex.Match(assets.tag_name, "[0-9]+\\.[0-9]+\\.[0-9]+").ToString();
+            MMP current = GetMMPVersion(Assembly.GetExecutingAssembly().GetName().Version.ToString());
+            MMP latest = GetMMPVersion(newestVersionSplit);
+
+            return (current.major != latest.major || current.minor != latest.minor || current.patch != latest.patch);
+        }
+        //returns the URL to the download link for the OS-specific version
+        private static string GetPath(GHUpdate assets, string OS)
+        {
+            for (int i = 0; i < assets.assets.Length; i++)
+            {
+                if (assets.assets[i].name.ToLower().Contains(OS.ToLower()))
+                {
+                    return assets.assets[i].browser_download_url;
+                }
+
+            }
+            return null;
+        }
+
+        public static async ValueTask Update()
+        {
+
+            GHUpdate assets = await GetGithubUpdates();
+
+            string newestVersionSplit = Regex.Match(assets.tag_name, "[0-9]+\\.[0-9]+\\.[0-9]+").ToString();
+            MMP latest = GetMMPVersion(newestVersionSplit);
+
+
+            //NOTE: linux is untested
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                string fileURL = GetPath(assets, "linux");
+                string fileName = "linux_" + latest.major + "." + latest.minor + "." + latest.patch + ".zip";
+                string directoryName = fileName + "temp";
+
+                //download and unzip to a temp directory
+                WebClient webClient = new WebClient();
+                webClient.DownloadFile(new Uri(fileURL), fileName);
+
+                if (Directory.Exists(directoryName))
+                {
+                    Directory.Delete(directoryName, true);
+                }
+
+                System.IO.Compression.ZipFile.ExtractToDirectory(fileName, directoryName);
+
+                //copy files from the temp directory to the root directory, then delete the temp directory and run the program again
+                ExecuteCommand("yes | cp -rf  \"" + System.IO.Directory.GetCurrentDirectory() + "/" + fileName + "temp\"" + "&& rm \"" + fileName + "\" && rm -rf \"" + System.IO.Directory.GetCurrentDirectory() + "/" + directoryName + "\"" + " && ./Lumper.UI");

```

Comment:
This is very hard to read. Let's use string interpolation (`$` at the start) plus (`@` at the start), pretty sure you can do both.
Also worth splitting each command into multiple lines.